File: 2328.java
import java.util.Scanner;
public class Main {
  static int a[];
  static String s[];
  public static boolean check(int length) {
    int i, n, m;
    n = a[length];
    for (i = 0; i < length; i++) {
      m = a[i] - n;
      if (m > 0 && !s[i].equals("high")) {
        return false;
      }
      if (m < 0 && !s[i].equals("low")) {
        return false;
      }
      if (m == 0 && !s[i].equals("on")) {
        return false;
      }
    }
    return true;
  }
  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);
    int i, n;
    while ((n = sc.nextInt()) != 0) {
      a = new int[20];
      s = new String[21];
      a[0] = n;
      i = 0;
      while (true) {
        sc.next();
        s[i] = sc.next();
        if (s[i].equals("on"))
          break;
        i++;
        a[i] = sc.nextInt();
      }
      if (check(i)) {
        System.out.println("Stan may be honest");
      } else {
        System.out.println("Stan is dishonest");
      }
    }
  }
}

File: 2328.cc
#include <iostream>
#include <cstdio>
#include <cstdlib>
#include <cstring>
using namespace std;

#define maxn 100

int f1[maxn], f2[maxn], count1, count2;

void make(int x, char *st)
{
    if (strcmp(st, "too high") == 0)
        f1[count1++] = x;
    else
        f2[count2++] = x;
}

bool check(int x)
{
    for (int i = 0; i < count1; i++)
        if (f1[i] <= x)
            return false;
    for (int i = 0; i < count2; i++)
        if (f2[i] >= x)
            return false;
    return true;
}

int main()
{
    int x;
    char st[30];
    while (scanf("%d", &x), x != 0)
    {
        count1 = 0;
        count2 = 0;
        getchar();
        gets(st);
        if (strcmp(st, "right on") == 0)
        {
            printf("Stan may be honest\n");
            continue;
        }
        make(x, st);
        while (1)
        {
            scanf("%d", &x);
            getchar();
            gets(st);
            if (strcmp(st, "right on") == 0)
                break;
            make(x, st);
        }
        if (check(x))
            printf("Stan may be honest\n");
        else
            printf("Stan is dishonest\n");
    }
    return 0;
}



